tests/classdefs.pyi

```diff
@@ -171,3 +173,13 @@ class BadGeneric(Generic[_T], int): ...  # Y059 "Generic[]" should always be the
 class BadGeneric2(int, typing.Generic[_T], str): ...  # Y059 "Generic[]" should always be the last base class
 class BadGeneric3(typing_extensions.Generic[_T], int, str): ...  # Y059 "Generic[]" should always be the last base class
 class BadGeneric4(Generic[_T], Iterable[int], str): ...  # Y059 "Generic[]" should always be the last base class
+
+class RedundantGeneric1(Iterable[_T], Generic[_T]): ...  # Y060 Redundant inheritance from "Generic[]"; class would be inferred as generic anyway
+class RedundantGeneric2(Generic[_S], GoodGeneric[_S]): ...  # Y059 "Generic[]" should always be the last base class  # Y060 Redundant inheritance from "Generic[]"; class would be inferred as generic anyway
+
+# Strictly speaking these inheritances from Generic are "redundant",
+# but people may consider it more readable to explicitly inherit from Generic,
+# so we deliberately don't flag them with Y060
+class GoodGeneric3(Container[_S], Iterator[_T], Generic[_S, _T]): ...
+class GoodGeneric4(int, Iterator[_T], str, Generic[_T]): ...
```

Comment:
I'd be OK with flagging this one and the next but don't feel strongly. The rule could be that we trigger only if there are exactly two bases that are `ast.Subscript`.

Reply:
Yeah, I had something like that in my initial implementation. But I reconsidered, as it started flagging classes like https://github.com/python/typeshed/blob/908993a807c77ce95c30b4815990dbb27fd49164/stubs/redis/redis/client.pyi#L79

I wasn't sure if we wanted to be flagging classes like that. There's an argument that it's easy to lose the fact that it's generic when you're skimming through the stubs if `CoreCommands[_StrType]` is buried in the middle of a long list of base classes